general/mem_manager.hpp

```diff
@@ -907,13 +918,20 @@ inline void Memory<T>::MakeAlias(const Memory &base, int offset, int size)
    h_mt = base.h_mt;
    h_ptr = base.h_ptr + offset;
    if (!(base.flags & REGISTERED))
-   { flags = (base.flags | ALIAS) & ~(OWNS_HOST | OWNS_DEVICE); }
-   else
    {
-      const size_t s_bytes = size*sizeof(T);
-      const size_t o_bytes = offset*sizeof(T);
-      MemoryManager::Alias_(base.h_ptr, o_bytes, s_bytes, base.flags, flags);
+      if (!MemoryManager::RegisterAliasBases())
+      {
+         flags = (base.flags | ALIAS) & ~(OWNS_HOST | OWNS_DEVICE);
+         return;
+      }
+      // Register 'base':
+      MemoryManager::Register_(base.h_ptr, nullptr, base.capacity*sizeof(T),
+                               base.h_mt, base.flags & OWNS_HOST,
+                               base.flags & ALIAS, base.flags);
```

Comment:
The following is a bit more clear I think:
```suggestion
      if (MemoryManager::RegisterAliasBases())
      {
         // Register 'base':
         MemoryManager::Register_(base.h_ptr, nullptr, base.capacity*sizeof(T),
                                  base.h_mt, base.flags & OWNS_HOST,
                                  base.flags & ALIAS, base.flags);
      }
      else
      {
         // copy the flags from 'base', setting the ALIAS flag to true, and setting both OWNS_HOST and OWNS_DEVICE to false
         flags = (base.flags | ALIAS) & ~(OWNS_HOST | OWNS_DEVICE);
      }
```

Reply:
Okay, I'll re-arrange this and add the suggested comment -- note that the `return;` statement still has to be there in the `else` block since in that case we want to skip the call to `MemoryManager::Alias_`.